python/triton/compiler/code_generator.py

```diff
@@ -289,6 +291,18 @@ class BoundJITMethod:
     __func__: JITFunction
 
 
+@dataclass
+class ValueDest:
+    ctx: Union[ast.Store, ast.Load]
+    value: base_value
+
+
+def _wrap_mutable_argument(arg, node):
+    if isinstance(arg, base_value) and getattr(arg.type, "__triton_mutable__", False):
+        setattr(arg, "__ast__", node)
```

Comment:
Will this cause issues with stacked function calls, e.g. `f(a)` calls `g(a)`.

Reply:
No, this works correct. Each subsequent call attaches the expression for its argument, and these are recursively propagated up to the last caller.